### ISSUE TITLE
Enhancement: Do not create locations/items for stuff not randomized via yaml options

### ISSUE THREAD

**Issue by ShinyNT** [WritingHusky/Twilight_Princess_apworld#60]
If you don't randomize bugs and/or poes, the generator screams this error:

[root at 2025-07-18 19:52:33,225]: Could not access required locations for accessibility check. Missing: [Agitha Female Ant Reward (TP), Agitha Female Beetle Reward (TP), Agitha Female Butterfly Reward (TP), Agitha Female Dayfly Reward (TP), Agitha Female Dragonfly Reward (TP), Agitha Female Grasshopper Reward (TP), Agitha Female Ladybug Reward (TP), Agitha Female Mantis Reward (TP), Agitha Female Phasmid Reward (TP), Agitha Female Pill Bug Reward (TP), Agitha Female Snail Reward (TP), Agitha Female Stag Beetle Reward (TP), Agitha Male Ant Reward (TP), Agitha Male Beetle Reward (TP), Agitha Male Butterfly Reward (TP), Agitha Male Dayfly Reward (TP), Agitha Male Dragonfly Reward (TP), Agitha Male Grasshopper Reward (TP), Agitha Male Ladybug Reward (TP), Agitha Male Mantis Reward (TP), Agitha Male Phasmid Reward (TP), Agitha Male Pill Bug Reward (TP), Agitha Male Snail Reward (TP), Agitha Male Stag Beetle Reward (TP), Jovani 20 Poe Soul Reward (TP), Jovani 60 Poe Soul Reward (TP)]

Which is both annoying to see in the AP console, but also useless and misleading and confusing, since the locations should not be "reachable" for goal anyway. Simply not creating the locations and items, based in the yaml option, would get rid of this.
